Sources/TuistCore/Graph/GraphTraverser.swift

```diff
@@ -572,6 +572,57 @@ public class GraphTraverser: GraphTraversing {
         return references
     }
 
+    public func needsEnableTestingSearchPaths(path: AbsolutePath, name: String) -> Bool {
+        var cache = NSCache<GraphTargetCacheKey, NSNumber>()
```

Comment:
Good point, I switched to a regular dictionary now. But instead of `Bool?` I went with non optional `Bool`, since I think the dictionary already encapsulates the `.none` case if the value to an entry cannot be found in the dictionary, it was not yet calculated. If it is found, it was already calculated and we can use the value right away. 

Does this fit better now ?

Reply:
It does! Thanks a lot for addressing the feedback @alexanderwe 🙏🏼